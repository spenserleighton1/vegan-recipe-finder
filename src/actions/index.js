export const addRecipes = (recipes) => ({
  type: 'ADD_RECIPES',
  recipes
})

export const searchByIngredient = (ingredients) => ({
  type: 'SEARCH_BY_INGREDIENT', 
  ingredients
})