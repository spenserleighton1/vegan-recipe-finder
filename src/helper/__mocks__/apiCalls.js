export const fetchRecipes = jest.fn().mockImplementation(() => ({
  recipes: [{title: 'a food'}, {title: 'more a food'}, {title: 'even more a food'}]
}));