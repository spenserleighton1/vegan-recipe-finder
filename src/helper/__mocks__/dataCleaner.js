export const cleanData = jest.fn();

export const cleanRecipe = jest.fn();