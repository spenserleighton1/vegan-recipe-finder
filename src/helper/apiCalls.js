export const fetchRecipes = async (key) => {
  const url = `https://food2fork.com/api/search?key=${key}&q=vegan`;
  const response = await fetch(url);
  const results = await response.json();
  return results;
}