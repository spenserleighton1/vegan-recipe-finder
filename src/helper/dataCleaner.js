export const cleanData = data => {
  const cleanData = data.map(recipe => {
    return {     
      recipeTitle: recipe.title,
      recipeLink: recipe.source_url,
      recipeImage: recipe.image_url,
      recipeId: recipe.recipe_id }
    })
  return cleanData;
}

export const cleanRecipe = data => ({
  title: data.recipe.title,
  publisher: data.recipe.publisher,
  ingredients: data.recipe.ingredients,
  linkUrl: data.recipe.source_url,
  id: data.recipe.recipe_id 
})
