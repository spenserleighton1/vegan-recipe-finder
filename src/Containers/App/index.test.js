import React from 'react';
import { shallow, mount } from 'enzyme';
import { App, mapDispatchToProps } from './';
import { addRecipes, isLoading } from '../../actions';
import { fetchRecipes } from '../../helper/apiCalls';
import { cleanData } from '../../helper/dataCleaner';

jest.mock('../../helper/apiCalls');
jest.mock('../../helper/dataCleaner');

describe('App', () => {
  it('should match the snapshot', () => {
    const mockAddRecipes = jest.fn();
    const mockIsLoading = jest.fn();
    const wrapper = shallow(<App isLoading={ mockIsLoading } addRecipes={ mockAddRecipes }/>)

    expect(wrapper).toMatchSnapshot()
  })

  it.skip('should call addRecipes on componentDidMount', async () => {
    const mockAddRecipes = jest.fn();
    const wrapper = shallow(<App addRecipes={ mockAddRecipes }/>);
     
    wrapper.instance().componentDidMount();
    await expect(fetchRecipes).toHaveBeenCalled();
  })

  describe('mapDispatchToProps', () => {
    it('should return propsObject', () => {
      const mockAddRecipes = jest.fn();
      const mockDispatch = jest.fn();

      const mappedProps = mapDispatchToProps(mockDispatch);
      const actionToDispatch = addRecipes([{title: 'food'}]);

      mappedProps.addRecipes([{title: 'food'}]);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  })
})
