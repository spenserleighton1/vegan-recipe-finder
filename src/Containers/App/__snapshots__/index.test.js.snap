// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addRecipes={[Function]}
    isLoading={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          authUser={null}
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/signUp"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/signIn"
/>,
        <Route
          exact={true}
          path="/savedRecipes"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/"
          render={[Function]}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "authUser": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/signUp",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/signIn",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/savedRecipes",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            authUser={null}
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/signUp"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/signIn"
/>,
          <Route
            exact={true}
            path="/savedRecipes"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "authUser": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/signUp",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/signIn",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/savedRecipes",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
